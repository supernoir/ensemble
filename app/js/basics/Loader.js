import React from 'react';
import intl from 'react-intl-universal';
import { Dimmer, Loader as SemanticLoader } from 'semantic-ui-react';

/**
 * Class Loader
 * component to be shown when a parent is loading
 */
export default class Loader extends React.Component {
	render(){
		return (
			<Dimmer active>
				<SemanticLoader inverted size='big'>{intl.get('app.loading')}</SemanticLoader>
			</Dimmer>
		);
	}
}