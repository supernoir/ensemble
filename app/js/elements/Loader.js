import React from 'react';
import intl from 'react-intl-universal';
import { Dimmer, Loader as SemanticLoader } from 'semantic-ui-react';

export default class Loader extends React.Component {
	render(){
		return (
			<Dimmer active>
				<SemanticLoader inverted size='big'>{intl.get('app.loading')}</SemanticLoader>
			</Dimmer>
		);
	}
}