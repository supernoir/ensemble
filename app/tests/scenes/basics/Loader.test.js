import React from 'react';
import { shallow } from 'enzyme';

// Import Components
import Loader from '../../../js/basics/Loader';

test('Loader -- Snapshot Test', () => {
	const wrapper = shallow(<Loader	/>
	);
	expect(wrapper).toMatchSnapshot();
});
