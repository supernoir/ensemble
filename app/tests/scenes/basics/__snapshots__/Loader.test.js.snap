// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader -- Snapshot Test 1`] = `
<Dimmer
  active={true}
>
  <Loader
    inverted={true}
    size="big"
  />
</Dimmer>
`;
